### DISCUSSION TITLE
Issues Creating Canadian Population Data

### DISCUSSION THREAD

**Discussion by pmedvedev** [synthetichealth/synthea#1532] [Q&A]
Hi all, I've checked a few threads that all seems to describe an error I'm seeing:

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "String.toUpperCase()" because the return value of "org.mitre.synthea.world.geography.Location.getAbbreviation(String)" is null
at org.mitre.synthea.world.agents.PayerManager.loadPayers(PayerManager.java:137)
at org.mitre.synthea.world.agents.PayerManager.loadPayers(PayerManager.java:107)
at org.mitre.synthea.engine.Generator.init(Generator.java:270)
at org.mitre.synthea.engine.Generator.(Generator.java:211)
at org.mitre.synthea.engine.Generator.(Generator.java:187)
at App.main(App.java:233)
```

I'm new to using Synthea. Trying my hand at generating some sample data but running into issues with the Canadian package. Looking for some guidance to documentation or general troubleshooting. I've cloned both main repo and international repo to my machine. I've ran "./gradlew build check test" and successfully created a patient with the standard US data. As a next step, I've followed the Readme guidance from the international package:

```
git clone https://github.com/synthetichealth/synthea
git clone https://github.com/synthetichealth/synthea-international
cd synthea-international
cp -R ca/* ../synthea
cd ../synthea
./run_synthea -p 5 Alberta
```

However, that did not work. I then assumed I need to rerun "./gradlew build check test", but unfortunately that also seemed to have failed. I've also tried clearing the insurance provider data as per https://github.com/synthetichealth/synthea/discussions/1464. That does end up generating some files but it's limited to practitioner and hospital information.

Appreciate any help/guidance in advance.

**Comment by jawalonoski** [comment]
The international files require some updates to match the latest version of the repository. I'm working on updating the files, slowly, in my limited spare time.

There is a work-in-progress `fixes` branch: https://github.com/synthetichealth/synthea-international/pull/20

None of changes currently address Canada, however.

I'll look into your specific error...

**Reply by pmedvedev** [reply]
First, thank you for the response! Could I try another working region and manually edit some of the reference data (e.g.: demographics)? Or should I fall back to an older version? At this stage, the data I'd like to generate is more proof of concept to see that directionally I can get a structure I need.

**Reply by jawalonoski** [reply]
If you just want to look at structure of the data, then just generate some US data and you can check the structure that way.

My guess is that running other Canadian provinces will have similar results as what you had with Alberta, but I haven't had a moment to verify that yet.

**Reply by pmedvedev** [reply]
Can I add some Canadian data to the standard data sources? I need to generate a patient located in Canada who's seen providers in Canada.

**Reply by jawalonoski** [reply]
I made some patches to the `ca/` subfolder of the `synthea-international` repository within the `fixes` branch...

After you clone the `synthea-international` repo, run:
```bash
git checkout fixes
cp -R ca/* ../synthea
```

You may have to run `git restore src/` within your `synthea` folder, to undo any changes from the previous files.

**Reply by pmedvedev** [reply]
Success! Thank you so much.

Deleted everything I had from before and ran:

```
git clone https://github.com/synthetichealth/synthea.git
git clone https://github.com/synthetichealth/synthea-international
cd synthea-international
git checkout fixes
cp -R ca/* ../synthea
cd ../synthea
./run_synthea -p 5 Alberta
```

Output:
2 -- Rocky100 Little434 (8 y/o M) Lac la Biche County, Alberta  (12053)
1 -- Rosie394 Rau926 (16 y/o F) Wood Buffalo, Alberta  (24217)
4 -- Orval846 Bechtelar572 (39 y/o M) Wood Buffalo, Alberta  (54352)
5 -- Lionel365 Ward668 (41 y/o M) Calgary, Alberta  (56671)
3 -- Elroy493 Wiza601 (69 y/o M) Red Deer, Alberta  (115174)
